### PR TITLE
Reset password

### PR DESCRIPTION
**Pull request by ZarinaPoyet**
CSS pending

### CODE REVIEW

File: server/controllers/users.ts [hunk 1]
```diff
@@ -340,3 +344,59 @@ export const logout = async (_req: Request, res: Response) => {
...
+      return res.status(404).json({ message: 'User not found' });
+    }
+
+    const resetToken = crypto.randomBytes(20).toString('hex');
+    user.resetPasswordToken = resetToken;
+    user.resetPasswordExpires = new Date(Date.now() + 3600000); // 1 hour
+
+    await user.save();
+
+    const resetLink = `http://localhost:5173/reset-password?token=${resetToken}`;
```
Comment (jianingroja) [inline server/controllers/users.ts:363]: use BASE_URL env variables

File: client/src/services/auth.ts [hunk 2]
```diff
@@ -85,3 +85,41 @@ export const registerPOST = async (
     console.error(error);
   }
 };
+
+export const sendPasswordResetEmail = async (email: string) => {
+  try {
+    const response = await fetch(`${BASE_URL}/forgot-password`, {
+      method: 'POST',
+      mode: 'cors',
+      credentials: 'include',
```
Comment (jianingroja) [inline client/src/services/auth.ts:94]: is this valid? at this point there is no credentials

File: client/src/hooks/useAuth.ts [hunk 3]
```diff
@@ -49,16 +49,37 @@ const useAuth = () => {
...
+      '/login',
+      '/register',
+      '/forgot-password',
+      '/reset-password',
+    ];
+
+    if (!name && !publicPaths.includes(pathname)) {
       navigate('/login');
-    } else if (pathname === '/') {
+      return;
```
Comment (jianingroja) [inline client/src/hooks/useAuth.ts:76]: `if ( publicPaths.includes(pathname) ) {
navigate(pathname)
} else if ( !name ) {
navigate('/login')
} else if ( pathname === '/' ){
navigate ('/feed')
} else {
navigate(pathname)
}`

File: client/src/services/auth.ts (hunk 2 above)
Comment (ZarinaPoyet) [reply client/src/services/auth.ts]: removed

**Review by jianingroja — APPROVED** [review]
lgtm

**Pull request merged**